src/power_grid_model/data_types.py

```diff
@@ -7,84 +7,210 @@
 have been defined and explained in this file.
 """
 
-from typing import TypeVar, Union
+from typing import TypeAlias, TypeVar, Union
 
 import numpy as np
 
-from power_grid_model.core.dataset_definitions import ComponentTypeLike  # pylint: disable=unused-import
 from power_grid_model.core.dataset_definitions import ComponentTypeVar
 
+AttributeType: TypeAlias = str
+"""
+An attribute type is a string reprenting the attribute type of a specific component.
+
+- Examples:
+
+    - "id"
+    - "u_rated"
+"""
+
+SparseDataComponentType: TypeAlias = str
+"""
+A string representing the component type of sparse data structures.
+
+Must be either "data" or "indptr".
+"""
+
 SingleArray = Union[np.ndarray]
 """
-A single array is a one-dimensional structured containing a list of components of the same type.
+A single array is a one-dimensional structured numpy array containing a list of components of the same type.
 
 - Examples:
 
     - structure: <1d-array>
     - concrete: array([(0, 10500.0), (0, 10500.0)], dtype=power_grid_meta_data["input"]["node"].dtype)
 """
 
+SingleColumn = Union[np.ndarray]
+"""
+A single column is a one-dimensional structured numpy array containing a list of the same attribute of
+multiple components of the same type.
+
+- Examples:
+
+    - structure: <1d-array>
+    - concrete:
+
+        - array([0, 1], dtype=power_grid_meta_data["input"]["node"].dtype.fields["id"][0])
+        - array([10500.0, 10500.0], dtype=power_grid_meta_data["input"]["node"].dtype.fields["u_rated"][0])
+"""
+
+SingleColumnarData = dict[AttributeType, SingleColumn]
+"""
+Single columnar data is a dictionary where the keys are the attribute types of the same component
+and the values are :class:`SingleColumn`.
+
+- Example: {"id": :class:`SingleColumn`, "u_rated": :class:`SingleColumn`}
+"""
+
 DenseBatchArray = Union[np.ndarray]
 """
 A dense batch array is a two-dimensional structured numpy array containing a list of components of 
-the same type for each scenario.
+the same type for each scenario. Otherwise similar to :class:`SingleArray`.
 """
 
-SparseBatchArray = dict[str, np.ndarray | SingleArray]
+BatchColumn = Union[np.ndarray]
+"""
+A batch column is a two-dimensional structured numpy array containing a list of the same attribute of
+multiple components of the same type. Otherwise, similar to :class:`SingleColumn`.
 """
-A sparse batch array is a dictionary containing the keys `indptr` and `data`.
 
-- data: a :class:`SingleArray`. The exact dtype depends on the type of component.
-- indptr: a one-dimensional numpy int64 array containing n+1 elements where n is the amount of scenarios.
+DenseBatchColumnarData = dict[AttributeType, BatchColumn]
+"""
+Batch columnar data is a dictionary where the keys are the attribute types of the same component
+and the values are :class:`BatchColumn`.
+
+- Example: {"id": :class:`BatchColumn`, "from_status": :class:`BatchColumn`}
```

Comment:
Should this be - Example: {"id": :class:`AttributeType`, "from_status": :class:`BatchColumn`}

Reply:
yes